internal/gcp/tags.go

```diff
@@ -178,3 +180,45 @@ func (m *tagsManager) GetProjectInfo(ctx context.Context, projectID string) (*re
 	m.cache.Set(cacheKey, project, tagCacheDuration)
 	return project, nil
 }
+
+func (m *tagsManager) DeleteValue(ctx context.Context, projectID string, key string, value string) error {
+
+	req := &resourcemanagerpb.DeleteTagValueRequest{
+		Name: value,
+	}
+
+	op, err := m.valuesClient.DeleteTagValue(ctx, req)
+	if err != nil {
+		return nil // Tag value already in use or deleted, consider this a success
+	}
+	_, err = op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to delete the tagValue %w", err)
+	}
+
+	m.cache.Delete(cacheKeyTagValue(key, value))
```

Comment:
I wonder what happens if the tag value is in the delete operation and then a new resources comes in that attaches that exact tag value. That might be quite difficult to test though, so I'm happy to just accept this as an edge case that we don't plan for (yet).

Reply:
We have tested this. The new tag key/value gets added but with a minimal delay.